src/rules/no-extraneous-dependencies.js

```diff
@@ -1,18 +1,14 @@
 import fs from 'fs'
 import path from 'path'
-import pkgUp from 'pkg-up'
+import readPkgUp from 'read-pkg-up'
 import minimatch from 'minimatch'
 import importType from '../core/importType'
 import isStaticRequire from '../core/staticRequire'
 
 function getDependencies(context) {
-  const filepath = pkgUp.sync(context.getFilename())
-  if (!filepath) {
-    return null
-  }
-
   try {
-    const packageContent = JSON.parse(fs.readFileSync(filepath, 'utf8'))
+    const pkg = readPkgUp.sync({cwd: context.getFilename(), normalize: false})
+    const packageContent = JSON.parse(pkg.pkg)
```

Comment:
The problem that makes the build fail is that `getDependencies` is now returning something else. `pkg.pkg` is now an object, and not a string that remains to be parsed. By attempting to parse it you are causing an error, which will be cached and make the function return `null`.

This works better:
```js
    const pkg = readPkgUp.sync({cwd: context.getFilename(), normalize: false})
    if (!pkg || !pkg.pkg) {
      return null
    }

    const packageContent = pkg.pkg
```